apps/studio/src/components/CoreTabs.vue

```diff
@@ -238,6 +263,48 @@
           }
         })
       },
+      completeDuplicateTableAction(){
+        const { tableName, schema,  entityType } = this.dbDuplicateTableParams
+
+        if (entityType !== 'table' && this.dbAction == 'duplicate') {
+          this.$noty.warning("Sorry, you can only duplicate tables.")
+          return;
+        }
+
+        if(tableName === this.duplicatedTableName){
+          this.$noty.warning("Sorry, you can't duplicate with the same name.")
+          return;
+        }
+
+        if (this.duplicatedTableName === null || this.duplicatedTableName === '' || this.duplicatedTableName === undefined) {
+          this.$noty.warning("Please enter a name for the new table.")
+          return;
+        }
+
+        this.$modal.hide(this.duplicateTableModal)
+
+        this.$nextTick(async() => {
+          try {
+            if (this.dbAction.toLowerCase() !== 'duplicate') {
+              return
+            }
+
+            await this.connection.duplicateTable(tableName, this.duplicatedTableName, schema)
```

Comment:
Do you need to clear duplicatesTableName and duplicatedTableParams here in a finally block?